runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/fieldwatch/fw001.c

```diff
@@ -0,0 +1,233 @@
+		return JNI_ERR;
+	}
+
+	memset(&capabilities, 0, sizeof(jvmtiCapabilities));
+	capabilities.can_generate_field_access_events = 1;  
+	capabilities.can_generate_field_modification_events = 1;
+	err = (*globalEnv)->AddCapabilities(globalEnv, &capabilities);
+	if (JVMTI_ERROR_NONE != err) {
+		printf("**ERROR**: Failed to AddCapabilities. JVMTI_ERROR code: %d.\n", err);
+		return JNI_ERR;
+	}
+
+	memset(&callbacks, 0, sizeof(jvmtiEventCallbacks));
+	callbacks.FieldAccess = fieldAccessCallback;
+	callbacks.FieldModification = fieldModificationCallback;
+	err = (*globalEnv)->SetEventCallbacks(globalEnv, &callbacks, sizeof(jvmtiEventCallbacks));
+	if (JVMTI_ERROR_NONE != err) {
+		printf("**ERROR**: Failed to set callbacks. JVMTI_ERROR code:  %d.\n", err);
+		return JNI_ERR;
+	}
+
+	err = (*globalEnv)->SetEventNotificationMode(globalEnv, JVMTI_ENABLE, JVMTI_EVENT_FIELD_ACCESS, NULL);
+	if (JVMTI_ERROR_NONE != err) {
+		printf("**ERROR**: Failed to enable field access event. JVMTI_ERROR code: %d.\n", err);
+		return JNI_ERR;
+	}
+
+	err = (*globalEnv)->SetEventNotificationMode(globalEnv, JVMTI_ENABLE, JVMTI_EVENT_FIELD_MODIFICATION, NULL);
+	if (JVMTI_ERROR_NONE != err) {
+		printf("**ERROR**: Failed to enable field modification event. JVMTI_ERROR code: %d.\n", err);
+		return JNI_ERR;
+	}
+
+	return JNI_OK;
+}
+
+/* Add class name to a global so we can invoke the appropriate Java API to update the number of reports seen. */
+JNIEXPORT jint JNICALL
+Java_com_ibm_jvmti_tests_fieldwatch_fw001_startTest(JNIEnv *jni_env, jobject cls, jstring name)
+{
+	testClassString = name;
+	testClassName = (*jni_env)->GetStringUTFChars(jni_env, name, NULL);
+	return JNI_OK;
+}
+
+JNIEXPORT jint JNICALL
+Java_com_ibm_jvmti_tests_fieldwatch_fw001_endTest(JNIEnv *jni_env, jobject cls)
+{
+	(*jni_env)->ReleaseStringUTFChars(jni_env, testClassString, testClassName);
+	return JNI_OK;
+}
+
+/* Add or remove field watches. */
+JNIEXPORT jint JNICALL
+Java_com_ibm_jvmti_tests_fieldwatch_fw001_modifyWatches(JNIEnv *env, jclass rcv, jstring className, jstring fieldName, jstring fieldSignature, jboolean isStatic, jboolean isRead, jboolean isWrite, jboolean isAdd)
+{
+	const char * _fieldName = (*env)->GetStringUTFChars(env, fieldName, NULL);
+	const char * _fieldSignature = (*env)->GetStringUTFChars(env, fieldSignature, NULL);
+	const char * _className = (*env)->GetStringUTFChars(env, className, NULL);
+	jfieldID fid;
+
+	if (NULL == _className) {
+		printf("_className string is NULL\n");
+		return JNI_ERR;
+	}
+
+	if (NULL == _fieldName) {
+		printf("_fieldName string is NULL\n");
+		return JNI_ERR;
+	}
+
+	if (NULL == _fieldSignature) {
+		printf("_fieldSignature string is NULL\n");
+		return JNI_ERR;
+	}
+
+	rcv = (*env)->FindClass(env, _className);
+	if (NULL == rcv) {
+		printf("Unable to find class with name: %s\n", _className);
+		return JNI_ERR;
+	}
+
+	if (isStatic) {
+		fid = (*env)->GetStaticFieldID(env, rcv, _fieldName, _fieldSignature);
+	}
+	else {
```

Comment:
Should be on previous line.

Reply:
...and more instances - please fix them all.

Reply:
This should be fixed now in 255c1d6